### PR TITLE
`DocumentQuestionAnsweringPipeline` only for fast ⚡ tokenizers

### PR DESCRIPTION
**Pull request by ydshieh**
# What does this PR do?

So let's make this more clear and explicit.

### CODE REVIEW

File: tests/models/layoutlm/test_modeling_layoutlm.py [hunk 1]
```diff
@@ -246,20 +246,6 @@ class LayoutLMModelTest(ModelTesterMixin, PipelineTesterMixin, unittest.TestCase
...
-        if (
-            pipeline_test_casse_name == "DocumentQuestionAnsweringPipelineTests"
-            and tokenizer_name is not None
-            and not tokenizer_name.endswith("Fast")
-        ):
-            # This pipeline uses `sequence_ids()` which is only available for fast tokenizers.
-            return True
-
-        return False
-
```
Comment (ydshieh) [inline tests/models/layoutlm/test_modeling_layoutlm.py:262]: See the change in `tests/test_pipeline_mixin.py`

File: tests/models/layoutlmv2/test_modeling_layoutlmv2.py [hunk 2]
```diff
@@ -295,15 +295,10 @@ def is_pipeline_test_to_skip(
...
-            and tokenizer_name is not None
-            and not tokenizer_name.endswith("Fast")
-        ):
-            # This pipeline uses `sequence_ids()` which is only available for fast tokenizers.
-            return True
 
-        return False
+        return super().is_pipeline_test_to_skip(
+            pipeline_test_casse_name, config_class, model_architecture, tokenizer_name, processor_name
+        )
```
Comment (ydshieh) [inline tests/models/layoutlmv2/test_modeling_layoutlmv2.py:301]: need to call super()

File: tests/test_pipeline_mixin.py [hunk 3]
```diff
@@ -428,9 +428,19 @@ def test_pipeline_zero_shot_image_classification(self):
...
     ):
+        # No fix is required for this case.
+        if (
+            pipeline_test_casse_name == "DocumentQuestionAnsweringPipelineTests"
+            and tokenizer_name is not None
+            and not tokenizer_name.endswith("Fast")
+        ):
+            # This pipeline uses `sequence_ids()` which is only available for fast tokenizers.
+            return True
+
```
Comment (ydshieh) [inline tests/test_pipeline_mixin.py:443]: A place where the logic doesn't rely on the model classes

**Review by Narsil — APPROVED** [review]
LGTM

**Review by sgugger — APPROVED** [review]
LGTM, thanks a lot!

**Pull request merged**